src/app/data-planes/data/index.ts

```diff
@@ -222,24 +223,36 @@ export const SidecarDataplane = {
     }
   },
 }
+
+type RuleConf = Exclude<InspectBaseRule['conf'], 'hostnames' | 'rules'> & {
+  hostnames: string[]
+  rules: ToTargetRefRule[]
+}
+
 export type Rule = Omit<InspectBaseRule, 'conf' | 'origin'> & {
   type: string
-  ruleType: string
+  ruleType: 'to' | 'from' | 'proxy'
   inbound?: InspectInbound
-  config: InspectBaseRule['conf']
+  config: RuleConf
   origins: InspectBaseRule['origin']
 }
 export type RuleCollection = Omit<PartialInspectRulesForDataplane, 'rules'> & {
   rules: Rule[]
 }
 export const Rule = {
-  fromObject(item: InspectBaseRule): Rule {
-    const { conf, origin, matchers, ...rest } = item
+  fromObject(item: InspectBaseRule, ruleType: 'to' | 'from' | 'proxy'): Rule {
+    const { conf = {}, origin, matchers, ...rest } = item
+
     return {
       type: '',
-      ruleType: '',
+      ruleType,
       ...rest,
-      config: Object.keys(conf || {}).length > 0 ? conf : {},
+      config: {
+        ...conf,
+        // An omitted or empty hostnames list implies the wildcard hostname (meaning any hostnames apply).
+        hostnames: Array.isArray(conf.hostnames) && conf.hostnames.length > 0 ? conf.hostnames : ['*'],
+        rules: Array.isArray(conf.rules) ? conf.rules : [],
+      },
```

Comment:
Isn't config the stuff we show in the UI? Shouldn't we avoid amending that? Or at least provide some sort of `raw` value that we don't touch to then display?

Reply:
Good point. Seeing as we’re now accessing specific data from a `conf` object, using `raw` becomes the preferred name for this, I think. I’ve changed this now and replaced the few instances of accessing `config` elsewhere to `raw` (for `Rule`, not for other objects that might still use `config`; we should change those, too).